source/PlayerInfo.cpp

```diff
@@ -2639,6 +2639,111 @@ void PlayerInfo::RegisterDerivedConditions()
+	// and any planetary storage.
+	auto &&presentOutfitProvider = conditions.GetProviderPrefixed("outfit (present): ");
+	presentOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		const Outfit *outfit = GameData::Outfits().Get(name);
+		const Planet *planet = GetPlanet();
+		if(planet)
+		{
+			retVal += Cargo().Get(outfit);
+			if(planetaryStorage.count(planet))
+				retVal += planetaryStorage[planet].Get(outfit);
+		}
+		for(const shared_ptr<Ship> &ship : ships)
+		{
+			if(ship->IsDestroyed() || ship->GetSystem() != system)
+				continue;
+			// If not on a planet, parked ships don't count. If on a planet, disabled ships don't count.
+			if((!planet && !ship->IsParked()) || (planet && !ship->IsDisabled()))
+			{
+				retVal += ship->OutfitCount(outfit);
+				retVal += ship->Cargo().Get(outfit);
+			}
+		}
+		return retVal;
+	});
+
+	// The following condition checks the player's entire fleet for installed outfits.
+	auto &&fleetOutfitProvider = conditions.GetProviderPrefixed("outfit (fleet installed): ");
+	fleetOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		const Outfit *outfit = GameData::Outfits().Get(name);
+		for(const shared_ptr<Ship> &ship : ships)
+		{
+			if(ship->IsDestroyed())
+				continue;
+			retVal += ship->OutfitCount(outfit);
+		}
+		return retVal;
+	});
+
+	// The following condition checks all cargo in the player's fleet.
+	auto &&fleetCargoOutfitProvider = conditions.GetProviderPrefixed("outfit (fleet cargo): ");
+	fleetCargoOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		const Outfit *outfit = GameData::Outfits().Get(name);
+		if(GetPlanet())
+			retVal += Cargo().Get(outfit);
+		for(const shared_ptr<Ship> &ship : ships)
+			if(!ship->IsDestroyed() && !ship->IsParked())
+				retVal += ship->Cargo().Get(outfit);
+		return retVal;
+	});
+
+	// The following condition checks the flagship's installed outfits.
+	auto &&flagshipOutfitProvider = conditions.GetProviderPrefixed("outfit (flagship installed): ");
+	flagshipOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		const Outfit *outfit = GameData::Outfits().Get(name);
+		if(flagship)
+			retVal += flagship->OutfitCount(outfit);
+		return retVal;
+	});
+
+	auto &&flagshipCargoOutfitProvider = conditions.GetProviderPrefixed("outfit (flagship cargo): ");
+	flagshipCargoOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		const Outfit *outfit = GameData::Outfits().Get(name);
+		if(GetPlanet())
+			retVal += Cargo().Get(outfit);
+		if(flagship)
+			retVal += flagship->Cargo().Get(outfit);
+		return retVal;
+	});
+
+	// The following condition checks planetary storage.
+	auto &&storedOutfitProvider = conditions.GetProviderPrefixed("outfit (storage): ");
```

Comment:
Would we want to differentiate between "storage on the currently landed upon planet", "storage in the current system" and "storage anywhere in the universe"?

Reply:
I don't really want to have to create a massive amount of prefixes to cover every possible location and combination, so how I've been thinking of this is in terms of what I might want to know that the player has access to, and in that sense there's really only three possible groups of locations to worry about:
* Is the outfit immediately accessible to the player? ("outfit (present)" or "outfit (flagship)", whether cargo or installed.)
* Is the outfit somewhere else, but the player could access it without leave the system? ("outfit (fleet)", whether cargo or installed. That is, if you launched and waited long enough, the ship with that outfit would reach you.)
* Is the outfit somewhere else, and the player would need to leave to get it? ("outfit (storage)")

In that sense, distinguishing between fleet installed and cargo might be unnecessary and so those two could be collapsed together, and storage could be changed to only check out-of-system storage. The only hole in this that I can think of is if you have something equipped on a ship, but that ship can't reach you, meaning it'd fit more into the third scenario of "the player would need to leave to get it."

Reply:
In the latest commit I added an `"outfit: "` prefix for checking all possible locations and an `"outfit (present storage)"` location for checking the current planet. I'm not sure what the usefulness of checking other planets in the current system would be, so I left that out.

Reply:
Good arguments. I'll might get back to this later (on consistency with other keywords and corner-cases), but the reasoning sounds good!